File: android/rctmgl/src/main/java/mapbox/rctmgl/events/RCTMGLEventTypes.java
package mapbox.rctmgl.events;

/**
 * Created by nickitaliano on 8/24/17.
 */

public class RCTMGLEventTypes {
    public static final String MAP_CLICK = "press";
    public static final String MAP_LONG_CLICK = "longpress";

    public static final String REGION_WILL_CHANGE = "regionwillchange";
    public static final String REGION_IS_CHANGING = "regionischanging";
    public static final String REGION_DID_CHANGE  = "regiondidchange";

    public static final String WILL_START_LOADING_MAP = "willstartloadingmap";
    public static final String DID_FINISH_LOADING_MAP = "didfinishloadingmap";
    public static final String DID_FAIL_LOADING_MAP = "didfailloadingmap";

    public static final String WILL_START_RENDERING_FRAME = "willstartrenderingframe";
    public static final String DID_FINISH_RENDERING_FRAME = "didfinishrenderingframe";
    public static final String DID_FINISH_RENDERING_FRAME_FULLY = "didfinishrenderingframefully";

    public static final String WILL_START_RENDERING_MAP = "willstartrenderingmap";
    public static final String DID_FINISH_RENDERING_MAP = "didfinishrenderingmap";
    public static final String DID_FINISH_RENDERING_MAP_FULLY = "didfinishrenderingmapfully";

    public static final String DID_FINISH_LOADING_STYLE = "didfinishloadingstyle";
}

File: ios/RCTMGL/RCTMGLEventTypes.m
//
//  RCTMGLEventTypes.m
//  RCTMGL
//
//  Created by Nick Italiano on 8/27/17.
//  Copyright © 2017 Mapbox Inc. All rights reserved.
//

#import "RCTMGLEventTypes.h"

@implementation RCTMGLEventTypes

NSString *const RCT_MAPBOX_EVENT_TAP = @"press";
NSString *const RCT_MAPBOX_EVENT_LONGPRESS = @"longpress";

NSString *const RCT_MAPBOX_USER_LOCATION_UPDATE = @"userlocationdupdated";
NSString *const RCT_MAPBOX_USER_TRACKING_MODE_CHANGE = @"usertrackingmodechange";

NSString *const RCT_MAPBOX_REGION_WILL_CHANGE_EVENT = @"regionwillchange";
NSString *const RCT_MAPBOX_REGION_IS_CHANGING = @"regionischanging";
NSString *const RCT_MAPBOX_REGION_DID_CHANGE = @"regiondidchange";

NSString *const RCT_MAPBOX_WILL_START_LOADING_MAP = @"willstartloadingmap";
NSString *const RCT_MAPBOX_DID_FINISH_LOADING_MAP = @"didfinishloadingmap";
NSString *const RCT_MAPBOX_DID_FAIL_LOADING_MAP = @"didfailoadingmap";

NSString *const RCT_MAPBOX_WILL_START_RENDERING_FRAME = @"willstartrenderingframe";
NSString *const RCT_MAPBOX_DID_FINSIH_RENDERING_FRAME = @"didfinishrenderingframe";
NSString *const RCT_MAPBOX_DID_FINISH_RENDERING_FRAME_FULLY = @"didfinishrenderingframefully";

NSString *const RCT_MAPBOX_WILL_START_RENDERING_MAP = @"willstartrenderingmap";
NSString *const RCT_MAPBOX_DID_FINISH_RENDERING_MAP = @"didfinishrenderingmap";
NSString *const RCT_MAPBOX_DID_FINISH_RENDERING_MAP_FULLY = @"didfinishrenderingmapfully";

NSString *const RCT_MAPBOX_DID_FINISH_LOADING_STYLE = @"didfinishloadingstyle";

NSString *const RCT_MAPBOX_ANNOTATION_TAP = @"annotationtap";

NSString *const RCT_MAPBOX_OFFLINE_PROGRESS = @"offlinestatus";
NSString *const RCT_MAPBOX_OFFLINE_ERROR = @"offlineerror";
NSString *const RCT_MAPBOX_OFFLINE_TILE_LIMIT = @"offlinetilelimit";

NSString *const RCT_MAPBOX_SHAPE_SOURCE_LAYER_PRESS = @"shapesourcelayerpress";
NSString *const RCT_MAPBOX_VECTOR_SOURCE_LAYER_PRESS = @"vectorsourcelayerpress";

@end
